postgresql/provider.go

```diff
@@ -130,17 +136,41 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	versionStr := d.Get("expected_version").(string)
 	version, _ := semver.Parse(versionStr)
 
-	config := Config{
-		Host:              d.Get("host").(string),
-		Port:              d.Get("port").(int),
-		Database:          d.Get("database").(string),
-		Username:          d.Get("username").(string),
-		Password:          d.Get("password").(string),
-		SSLMode:           sslMode,
-		ApplicationName:   tfAppName(),
-		ConnectTimeoutSec: d.Get("connect_timeout").(int),
-		MaxConns:          d.Get("max_connections").(int),
-		ExpectedVersion:   version,
+	var config Config
+	connString := d.Get("connection_string").(string)
+
+	if connString != "" {
+		conn, err := parseConnectionString(connString)
```

Comment:
Instead of writing our own connection string parsing routine, can you import `jackc/pgx` and use its [`ParseConnectionString()`](https://godoc.org/github.com/jackc/pgx#ParseConnectionString) routine instead?  Make two commits: the first uses the new library, the second is the vendor import of `jackc/pgx`.  If you're not comfortable doing this, let me know and I'll tackle it.

Similarly, if you wanted to map the existence of a non-empty `dsn` string to [`ParseDSN()`](https://godoc.org/github.com/jackc/pgx#ParseDSN), that'd be a +1 for usability for some people.